Update topdown_gru.py

ForeverLMZ opened the pull request:
adapted topdown feature

ForeverLMZ committed:
Update topdown_gru.py

adapted topdown feature

ForeverLMZ committed:
Update training.py

masht18 reviewed (commented):
I'm not sure why we need Line 141.  Is it because of the hack I mentioned using to implement topdown before I had feedback linears? Either way, there's no need for that hack now. Don't change anything within a single ConvGRU block.

ForeverLMZ committed:
Fixed time sequence dimension

ForeverLMZ committed:
buggy version of modular topdown

ForeverLMZ committed:
minor changes

ForeverLMZ committed:
Create graph.py

ForeverLMZ committed:
Minor Updates after meeting

ForeverLMZ committed:
Update graph.py

ForeverLMZ committed:
graph to architecture code blueprint

ForeverLMZ committed:
minor fix

ForeverLMZ committed:
Update graph.py

ForeverLMZ committed:
it runs

ForeverLMZ committed:
it runs, now with multiple inputs + adapted for diff dim/size

ForeverLMZ committed:
chance level accuracy

ForeverLMZ committed:
Fixed training error from not-stored-linear projection layers

ForeverLMZ committed:
Update graph.py

Changed bottomup_gru_list and topdown_gru_list to 1D list (got rid of the time dimension). Padded the bottomup and modulatory signal with zeros to fit the same size of (before reshaping) (b, n*hidden_dim, h, w) -> (32, 40, 28,28). Now training accuracy approaches 100% and loss approaches 0%, yet val accuracy remains at ~20%. Serious overfitting problem

ForeverLMZ committed:
works for single node model

added initializations for node rank_list
